java/maven/src/main/java/net/sf/mmm/code/java/maven/impl/MavenBridgeImpl.java

```diff
@@ -213,25 +213,33 @@ public Model readEffectiveModelFromLocation(File location, boolean fallback) {
 
   private Properties resolveProperties(Properties properties, File projectBaseDir) {
 
-    try {
-      File mvnDir = new File(projectBaseDir, ".mvn");
-      if (mvnDir.isDirectory()) {
-        File mvnConfig = new File(mvnDir, "maven.config");
-        if (mvnConfig.isFile()) {
-          properties = new Properties(properties);
-          List<String> lines = Files.readAllLines(mvnConfig.toPath());
-          for (String line : lines) {
-            resolveProperties(properties, line.trim());
+    if (projectBaseDir.getParentFile() == null) {
+      return properties;
+    }
+
+    boolean pomFileExists = new File(projectBaseDir, POM_XML).exists();
+    boolean parentPomFileExists = new File(projectBaseDir.getParentFile(), POM_XML).exists();
+    if (pomFileExists && !parentPomFileExists) {
```

Comment:
these ifs with duplicated conditions are coded a little bit confusing.
Further my thought was that `.mvn/maven.config` should be present in "projects's top level directory" while your code seems to do the oposite.
Is the `!` inversed?

Reply:
It's the check that we ended at the top-level. No pom.xml exists at the current directory, whereas there is no pom.xml in the next parent directory. Therefore, throwing the recursion anchor here.

Reply:
But you were right, I have overseen the first case, which didn't make much sense. Corrected in last commit.